contracts/utils/OrderPayable.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../access-control-registry/AccessControlRegistryAdminnedWithManager.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+contract OrderPayable is AccessControlRegistryAdminnedWithManager {
+    using ECDSA for bytes32;
+
+    event PaidForOrder(
+        bytes32 indexed orderId,
+        uint256 expirationTimestamp,
+        address orderSigner,
+        uint256 amount
+    );
+
+    event PaidInvoice(
+        bytes32 indexed orderId,
+        uint256 expirationTimestamp,
+        address orderSigner,
+        address invoicePayer,
+        uint256 amount
+    );
+
+    event Withdrew(address recipient, uint256 amount);
+
+    string public constant ORDER_SIGNER_ROLE_DESCRIPTION = "Order signer";
+    string public constant WITHDRAWER_ROLE_DESCRIPTION = "Withdrawer";
+    bytes32 public immutable orderSignerRole;
+    bytes32 public immutable withdrawerRole;
+
+    mapping(bytes32 => bool) public orderIdToPaymentStatus;
+
+    modifier onlyValidOrderParameters(
+        bytes32 orderId,
+        uint256 expirationTimestamp,
+        address orderSigner,
+        uint256 paymentAmount
+    ) {
+        require(orderId != bytes32(0), "Order ID zero");
+        require(expirationTimestamp > block.timestamp, "Order expired");
+        require(
+            orderSigner == manager ||
+                IAccessControlRegistry(accessControlRegistry).hasRole(
+                    orderSignerRole,
+                    orderSigner
+                ),
+            "Invalid order signer"
+        );
+        require(paymentAmount > 0, "Payment amount zero");
+        require(!orderIdToPaymentStatus[orderId], "Order already paid for");
+        _;
+    }
+
+    constructor(
+        address _accessControlRegistry,
+        string memory _adminRoleDescription,
+        address _manager
+    )
+        AccessControlRegistryAdminnedWithManager(
+            _accessControlRegistry,
+            _adminRoleDescription,
+            _manager
+        )
+    {
+        orderSignerRole = _deriveRole(
+            _deriveAdminRole(manager),
+            ORDER_SIGNER_ROLE_DESCRIPTION
+        );
+        withdrawerRole = _deriveRole(
+            _deriveAdminRole(manager),
+            WITHDRAWER_ROLE_DESCRIPTION
+        );
+    }
+
+    function payForOrder(
+        bytes32 orderId,
+        uint256 expirationTimestamp,
+        address orderSigner,
+        bytes calldata signature
```

Comment:
If we encode this into a single `bytes` it makes the multisig user flow simpler

Reply:
Done [here](https://github.com/api3dao/airnode-protocol-v1/pull/128/commits/bbe39cf7a77a19953e110c3bf492b871382d634a)